packages/ref-metrics-esmvaltool/src/cmip_ref_metrics_esmvaltool/metrics/tcre.py

```diff
@@ -0,0 +1,125 @@
+
+    experiments = (
+        "esm-1pctCO2",
+        "esm-piControl",
+    )
+    variables = (
+        "tas",
+        "fco2antt",
+    )
+    data_requirements = (
+        DataRequirement(
+            source_type=SourceDatasetType.CMIP6,
+            filters=(
+                FacetFilter(
+                    facets={
+                        "variable_id": variables,
+                        "frequency": "mon",
+                        "experiment_id": experiments,
+                    },
+                ),
+                FacetFilter(
+                    facets={
+                        "variable_id": "fco2antt",
+                        "experiment_id": "esm-piControl",
+                    },
+                    keep=False,
+                ),
+            ),
+            group_by=("source_id", "member_id", "grid_label"),
+            constraints=(
+                RequireFacets("experiment_id", experiments),
+                RequireFacets("variable_id", variables),
+                RequireContiguousTimerange(group_by=("instance_id",)),
+                RequireOverlappingTimerange(group_by=("instance_id",)),
+                AddSupplementaryDataset.from_defaults("areacella", SourceDatasetType.CMIP6),
+            ),
+        ),
+    )
+
+    @staticmethod
+    def update_recipe(recipe: Recipe, input_files: pandas.DataFrame) -> None:
+        """Update the recipe."""
+        # Prepare updated datasets section in recipe. It contains three
+        # datasets, "tas" and "fco2antt" for the "esm-1pctCO2" and just "tas"
+        # for the "esm-piControl" experiment.
+        recipe_variables = dataframe_to_recipe(input_files)
+        tas_esm_1pctCO2 = next(
+            ds for ds in recipe_variables["tas"]["additional_datasets"] if ds["exp"] == "esm-1pctCO2"
+        )
+        fco2antt_esm_1pctCO2 = next(
+            ds for ds in recipe_variables["fco2antt"]["additional_datasets"] if ds["exp"] == "esm-1pctCO2"
+        )
+        tas_esm_piControl = next(
+            ds for ds in recipe_variables["tas"]["additional_datasets"] if ds["exp"] == "esm-piControl"
+        )
+        tas_esm_piControl["timerange"] = tas_esm_1pctCO2["timerange"]
+
+        recipe["diagnostics"]["tcre"]["variables"] = {
+            "tas_esm-1pctCO2": {
+                "short_name": "tas",
+                "preprocessor": "global_annual_mean_anomaly",
+                "additional_datasets": [tas_esm_1pctCO2],
+            },
+            "tas_esm-piControl": {
+                "short_name": "tas",
+                "preprocessor": "global_annual_mean_anomaly",
+                "additional_datasets": [tas_esm_piControl],
+            },
+            "fco2antt": {
+                "preprocessor": "global_cumulative_sum",
+                "additional_datasets": [fco2antt_esm_1pctCO2],
+            },
+        }
+        recipe["diagnostics"].pop("barplot")
+
+    @staticmethod
+    def format_result(result_dir: Path) -> OutputBundle:
```

Comment:
We now have concrete types `cmip_ref_core.pycmec.output.CMECOutput` and `cmip_ref_core.pycmec.output.CMECMetric` instead of this loose type hint.
The content here is also representative of a metrics bundle, not an output bundle. 

The output bundle contains references to plots and datafiles, where as the metric bundle contains the calculated scalars. The output bundle content is not currently implemented for ESMValTool (see `packages/ref-metrics-pmp/src/cmip_ref_metrics_pmp/pmp_driver.py` for an example of including data and plots in an output bundle).

I'd recommend fixing this up in a follow-up PR as is will impact all of the ESMValTool metrics